README.md

```diff
@@ -438,6 +438,30 @@ DeltaHelpers.deltaNumRecordDistribution(path)
 +------------------------------------------------+--------------------+------------------+--------------------+--------------------+------------------+---------------------------------------------------------+
 ```
 
+## Number of Shuffle Files in Merge & Other Filter Conditions
+The function `getNumShuffleFiles` gets the number of shuffle files (part files for parquet) that will be pulled into memory for a given filter condition. This is particularly useful in a Delta Merge operation where the number of shuffle files can be a bottleneck. 
+Running the merge condition through this method can give an idea about the amount of memory resources required to run the merge.
+For example, if the condition is "country = 'GBR' and age >= 30 and age <= 40 and firstname like '%Jo%' " and country is the partition column, 
+```scala
+DeltaHelpers.getNumShuffleFiles(path, "country = 'GBR' and age >= 30 and age <= 40 and firstname like '%Jo%' ")
+```
+
+then the output might look like 
+```scala
+(18, 100, 300, 600, 800, 800, List())
```

Comment:
Should this output be a map that indicates what each value represents?

Reply:
I have changed the output to a MAP and updated the method comments. Please check if it looks good, then I will update README.md @MrPowers